tests/property/test_predicates.py

```diff
@@ -46,6 +47,7 @@ def test_equals_has_expected_repr():
 
 @settings(max_examples=200)
 @given(type_strs)
+@example('int0')
```

Comment:
Is `int0` a valid type`?  in fact is anything with a sub of `0` valid?

Reply:
No, it's not.  But it's parsable.  I'm pretty sure it could be parsed by the regexes we were using as well.  I guess I could update the grammar so that "0" is not a parsable sub component.  But it seemed like we were already drawing a pretty clear line between parsability and validity.